### PR TITLE
feat(DatePicker): add function support for format

### PR DESCRIPTION
**Pull request by kerm1it**
[#17222](https://github.com/ant-design/ant-design/issues/17222)

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Solution

add format function support for DatePicker

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/utils.ts [hunk 1]
```diff
@@ -2,13 +2,18 @@ import * as moment from 'moment';
...
   if (!value) {
     return '';
   }
   if (Array.isArray(format)) {
     format = format[0];
   }
+
+  if (typeof format === 'function') {
+    format = format(value);
+  }
```
Comment (afc163) [inline components/date-picker/utils.ts]: You should start with http://github.com/react-component/calendar

Comment (Kermit-Xuan) [reply components/date-picker/utils.ts]: 👌

Comment (Kermit-Xuan) [reply components/date-picker/utils.ts]: [#599](https://github.com/react-component/calendar/pull/599)

**Comment by afc163** [comment]
conflict and ci failed

**Pull request closed without merging**